gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/subject/SubjectManagerV2.java

```diff
@@ -55,7 +56,7 @@ public String generateSubFrom(User user) {
 
     @Override
     public String generateInternalSubFrom(User user) {
-        return user.getSource() + SEPARATOR + user.getExternalId();
+        return user.getFullId().getInternalSubject();
```

Comment:
**issue**: I don't think it is the responsabilty of the UserId object as for now it is dependant of the Domain version.